webrtc/browser-to-browser.md

```diff
@@ -0,0 +1,130 @@
+
+## Open Questions
+
+- Do we need a mechanism for browsers to advertise support for WebRTC browser-to-browser?
+
+  Say that browser B supports WebRTC browser-to-browser.
+  B listens via a relay and advertises its relayed address.
+  A discovers B's relayed address.
+  At this point A does not know whether B is a browser and thus supports WebRTC browser-to-browser, or whether B is e.g. a laptop potentially supporting TCP and QUIC hole punching via DCUtR but not WebRTC browser-to-browser.
+  In the latter case, A can not establish a direct connection to B.
+
+  Potential solution would be for B to advertise some protocol after the `/p2p-circuit` within its Multiaddr, e.g. `/ip6/<RELAY_IP>/udp/4001/p2p/<RELAY_PEER_ID>/p2p-circuit/webrtc-direct/p2p/<B_PEER_ID>`.
+  As an alternative, A can discover B's support via the identify protocol on the relayed connection or by optimistically opening a stream using the signaling protocol.
+  Both of the latter options imply long latency (direct connection + relayed connection + stream establishment / identify exchange) on success and on failure happen at the expense of a wasted relayed connection.
+
+## FAQ
+
+- Why is there no additional Noise handshake needed?
+
+  This specification (browser-to-browser) requires _A_ and _B_ to exchange their SDP offer and answer over an authenticated channel.
+  Offer and answer contain the TLS certificate fingerprint.
+  The browser validates the TLS certificate fingerprint through the DTLS handshake during the WebRTC connection establishment.
+
+  In contrast, the browser-to-server specification allows exchange of the server's multiaddr, containing the server's TLS certificate fingerprint, over unauthenticated channels.
+  In other words, the browser-to-server specification does not consider the TLS certificate fingerprint in the server's multiaddr to be trusted.
+
+- Why use a custom signaling protocol? Why not use [DCUtR]?
+
+  DCUtR offers time synchronization through a two-step protocol (first `Connect`, then `Sync`).
+  This is not needed for WebRTC.
+
+  DCUtR does not provide a mechanism to trickle local address candidates to the remote as they are discovered.
+  Trickling candidates just-in-time allows for faster WebRTC connection establishment.
+
+- Why does _A_ and not _B_ initiate the signaling protocol?
+
+  In [DCUtR] _B_ (inbound side of the relayed connection) initiates the [DCUtR] protocol by opening the [DCUtR] protocol stream.
+  The reason is that in case _A_ is publicly reachable, _B_ might be able to use connection reversal to connect to _A_ directly.
+  This reason does not apply to the WebRTC browser-to-browser protocol.
+  Given that _A_ and _B_ at this point already have a relayed connection established, they might as well use it to exchange SDP, instead of using connection reversal and WebRTC browser-to-server.
+  Thus, for the WebRTC browser-to-browser protocol, _A_ initiates the signaling protocol by opening the signaling protocol stream.
```

Comment:
@ckousik @MarcoPolo as just discussed in the call, e99d582 changes the specification to have _A_ initiating the signaling protocol by opening the signaling stream.